models/base/scratch/bigquery/snowplow_web_base_events_this_run.sql

```diff
@@ -0,0 +1,45 @@
+{{ 
+  config(
+    materialized='table',
+    partition_by = {
+      "field": "collector_tstamp",
+      "data_type": "timestamp",
+      "granularity": "day"
+    },
+    cluster_by=["event_name","page_view_id"],
+    tags=["this_run"]
+  ) 
+}}
+
+{%- set lower_limit, upper_limit = snowplow_utils.return_limits_from_model(ref('snowplow_web_base_sessions_this_run'),
+                                                                          'start_tstamp',
+                                                                          'end_tstamp') %}
+
+-- without downstream joins, it's safe to dedupe by picking the first event_id found.
+select
+  array_agg(e order by e.collector_tstamp limit 1)[offset(0)].*
+
+from (
+
+  select
+    a.contexts_com_snowplowanalytics_snowplow_web_page_1_0_0[safe_offset(0)].id as page_view_id,
+    b.domain_userid, -- take domain_userid from manifest. This ensures only 1 user id per session.
```

Comment:
Nit: I think one of the below would make this comment a bit clearer:

This ensures only 1 id per session.
This ensures only 1 domain user id per session.

Reply:
Ps. Nice neat solution to the issue!

Reply:
Yeh fair 'user id' is a bit ambiguous